### ISSUE TITLE
web-wallet: Improve experience when invalid Gas Limit is set on initialization and the user accesses Settings

### ISSUE THREAD

**Issue by nortonandreev** [dusk-network/rusk#1558]
#### Summary

I am seeing a problem with the experience, in case an invalid gas limit has been set in the ENV.

<img width="736" alt="Screenshot 2024-03-18 at 13 46 21" src="https://github.com/dusk-network/rusk/assets/34094428/140ab3ec-7966-4382-9aea-dee58c90a552">

If the user tries to access the Settings, the Back button will be disabled, as the Gas Limit is invalid. However, as the user hasn't changed anything in the view, I think it's not immediately clear why the Back is disabled.

This is more of a corner case, as the user has to have supplied an invalid value in their ENV. However, it is possible that the ENV gets messed up on Production, which would lead to users experiencing this.

Open to discussion.

**Comment by nortonandreev** [comment]
> You make a valid point here, but still unsure if we should act on it now, or rather wait for some change I'm expecting like contracts defining their own gas settings limits (upper / lower) and defaults.

Sure, it's a low-priority item. Just wanted to track it.

**Comment by deuch13** [comment]
This is actually a feature not a bug, I made it like this for the following edge case. 
We or the user change the ENVs which invalidate the preferred gas settings in the app. As we do not show that the ENVs have changed the only way to make the user change the preferred gas settings is to force him to change them when he goes to the settings page.

**Comment by nortonandreev** [comment]
> This is actually a feature not a bug, I made it like this for the following edge case. We or the user change the ENVs which invalidate the preferred gas settings in the app. As we do not show that the ENVs have changed the only way to make the user change the preferred gas settings is to force him to change them when he goes to the settings page.

I don't think this is true, the user will be forced to change them once they trigger a flow that makes use of the properties.

<img width="697" alt="Screenshot 2024-03-19 at 14 20 14" src="https://github.com/dusk-network/rusk/assets/34094428/79795db5-29b3-4b37-8396-cbbc362d478e">

And progressing to the next step being disabled due to invalid value is well-known behavior, whereas disabling the back on the Settings doesn't feel intuitive.


---
Side note: I think that, if this is the case, the Gas Settings pane should be expanded by default, otherwise the issue is not immediately clear as well.

<img width="638" alt="Screenshot 2024-03-19 at 14 20 42" src="https://github.com/dusk-network/rusk/assets/34094428/255e98ed-f77b-4ec5-8149-4ac8e721b52b">

**Comment by deuch13** [comment]
> > This is actually a feature not a bug, I made it like this for the following edge case. We or the user change the ENVs which invalidate the preferred gas settings in the app. As we do not show that the ENVs have changed the only way to make the user change the preferred gas settings is to force him to change them when he goes to the settings page.
> 
> The user will be forced to change them once they trigger a flow that makes use of the properties.
> 
> <img alt="Screenshot 2024-03-19 at 14 20 14" width="697" src="https://private-user-images.githubusercontent.com/34094428/314049900-79795db5-29b3-4b37-8396-cbbc362d478e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4NTU2MjYsIm5iZiI6MTcxMDg1NTMyNiwicGF0aCI6Ii8zNDA5NDQyOC8zMTQwNDk5MDAtNzk3OTVkYjUtMjliMy00YjM3LTgzOTYtY2JiYzM2MmQ0NzhlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE5VDEzMzUyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMyNjY0YTlhNzYyYWY2N2NjMTdhNjA2ZTVlYzcyMGE0NzRkM2M3MDQ3NDNiMDkwMGY5OWJjYzA0NDRhMGViZTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OOccYHKBLJbLYz1fuyjyRWOvcc29YZJt1ey_N91BpuA">
> And progressing to the next step being disabled due to invalid value is a well-known behavior, whereas disabling the back on the Settings doesn't feel intuitive.
>
Indeed, but I do not know what the user will do first in the app so that is why I forced the user in the settings page to update the gas limits.

> Side note: I think that, invalid values for the Gas have been set for whatever reason, the Gas Settings pane should be expanded by default, otherwise why the user cannot proceed is not immediately clear.
> 
> <img alt="Screenshot 2024-03-19 at 14 20 42" width="638" src="https://private-user-images.githubusercontent.com/34094428/314050176-255e98ed-f77b-4ec5-8149-4ac8e721b52b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4NTU2MjYsIm5iZiI6MTcxMDg1NTMyNiwicGF0aCI6Ii8zNDA5NDQyOC8zMTQwNTAxNzYtMjU1ZTk4ZWQtZjc3Yi00ZWM1LTgxNDktNGFjOGU3MjFiNTJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE5VDEzMzUyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYxMzRjMmJhOGZlMmY3Y2E5NGY4Yzg0ZjU5ZmZkMmQ4Y2VmMDA4ZTM4NDc2ODQzZjRjYjIzNTQzZDJiZWUyNWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JB6d1C30ifKvS3ed5PMMxLsRMKTS4pI_HHbibS725wU">

This is strange as the pane should open if the gas limits are invalid, this indeed is a bug.

**Comment by nortonandreev** [comment]
> This is strange as the pane should open if the gas limits are invalid, this indeed is a bug.

I just checked, it seems to be working fine. Not sure what happened before.
